### PR TITLE
Include setup_docker_repo in docker_service and allow old docker-ce v…

### LINKED ISSUE #1014: docker_installation_package fails to install 18.06.0.ce-3.el7 on centos 7
### Cookbook version
4.4.1

### Chef-client version
13.10.0

### Platform Details
Centos 7.4

### Scenario:
when only the version property is passed into this resource for 18.06.0.ce-3.el7, package_version is determined by `#version_string` in the resource. This fails to install the correct version, because https://github.com/chef-cookbooks/docker/blob/master/libraries/docker_installation_package.rb#L157 hardcodes the trailing part of the version string.

### Steps to Reproduce:
The yum repo is configured by the [chef-yum-docker](https://github.com/chef-cookbooks/chef-yum-docker) cookbook in our scenario.

```
docker_installation_package 'docker-ce' do
  setup_docker_repo false
  version 18.06.0
end
```

### Expected Result:
This is a bit tricky, in that one would never know ahead of time what series in the release of the specified version is available. The `18.06.0.ce-1` version doesn't seem to exist in the yum repo anymore, so it would need to look at all available versions of that series and pick one (maybe the latest).

 
### Actual Result:
```
           ================================================================================
           Error executing action `create` on resource 'docker_installation_package[docker-ce]'
           ================================================================================
           
           Chef::Exceptions::Package
           -------------------------
           yum_package[docker-ce] (/tmp/kitchen/cache/cookbooks/docker/libraries/docker_installation_package.rb line 42) had an error: Chef::Exceptions::Package: Version ["18.06.0.ce-1.el7.centos"] of ["docker-ce"] not found. Did you specify both version and release? (version-release, e.g. 1.84-10.fc6)
           
           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/my_docker/recipes/docker.rb
           
            76: docker_installation_package node['my_docker']['docker']['package_name'] do
            77:   setup_docker_repo false
            78:   # package_version node['my_docker']['docker']['package_version']
            79:   version node['my_docker']['docker']['version']
            80: end
            81: 
           
           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/my_docker/recipes/docker.rb:76:in `from_file'
           
           docker_installation_package("docker-ce") do
             action [:create]
             default_guard_interpreter :default
             declared_type :docker_installation_package
             cookbook_name "my_docker"
             recipe_name "docker"
             version "18.06.0"
             package_name "docker-ce"
             package_version "18.06.0.ce-1.el7.centos"
           end
```

Workaround
=========
instead of using the `version` property, use the `package_version` property and be very specific about it, such as:
```
docker_installation_package 'docker-ce' do
  setup_docker_repo false
  package_version '18.06.0.ce-3.el7'
end
```
Using package_version installs the version properly, as follows:
```
         * docker_installation_package[docker-ce] action create[2018-07-28T19:18:28+00:00] INFO: Processing docker_installation_package[docker-ce] action create (my_docker::docker line 76)
       [2018-07-28T19:18:28+00:00] INFO: Processing docker_installation_package[docker-ce] action create (my_docker::docker line 76)
       
           * yum_package[docker-ce] action install[2018-07-28T19:18:28+00:00] INFO: Processing yum_package[docker-ce] action install (/tmp/kitchen/cache/cookbooks/docker/libraries/docker_installation_package.rb line 42)
       [2018-07-28T19:18:28+00:00] INFO: Processing yum_package[docker-ce] action install (/tmp/kitchen/cache/cookbooks/docker/libraries/docker_installation_package.rb line 42)
       [2018-07-28T19:18:28+00:00] INFO: yum_package[docker-ce] installing docker-ce-18.06.0.ce-3.el7 from docker-stable repository
       [2018-07-28T19:18:28+00:00] INFO: yum_package[docker-ce] installing docker-ce-18.06.0.ce-3.el7 from docker-stable repository
```

### PR DESCRIPTION
**Pull request by ChiefAlexander**
…ersions for centos

Signed-off-by: Alex Harder <Alex.Harder@Cerner.com>

### Description

Includes setup_docker_repo within docker_service
Allows older docker-ce versions to be used for centos

### Issues Resolved

Fixes #1014 

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: libraries/docker_installation_package.rb [hunk 1]
```diff
@@ -128,9 +128,11 @@ def version_string(v)
                 end
 
       # https://github.com/seemethere/docker-ce-packaging/blob/9ba8e36e8588ea75209d813558c8065844c953a0/deb/gen-deb-ver#L16-L20
-      test_versioning = '3'
+      test_versioning = new_resource.version.to_f > 18.03 ? '3' : '1'
```
Comment (chasebolt) [inline libraries/docker_installation_package.rb]: change `new_resource.version.to_f` to `v.to_f`

File: libraries/docker_installation_package.rb [hunk 2]
```diff
@@ -128,9 +128,11 @@ def version_string(v)
...
 
-      return "#{v}#{edition}-#{test_versioning}.el7" if el7?
+      centos_extra = new_resource.version.to_f > 18.03 ? '' : '.centos'
```
Comment (chasebolt) [inline libraries/docker_installation_package.rb]: change `new_resource.version.to_f` to `v.to_f`

File: libraries/docker_installation_package.rb (hunk 1 above)
Comment (ChiefAlexander) [reply libraries/docker_installation_package.rb]: I completely missed that we where already passing this in 🤦‍♂️

**Comment by chasebolt** [comment]
thanks! released in v4.6.1

**Pull request merged**